gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireContainer.java

```diff
@@ -1931,7 +1931,7 @@ final void drop(final GemFireTransaction tran) throws StandardException {
           && (indexManager = (GfxdIndexManager)r.getIndexUpdater()) != null) {
         indexManager.drop(tran);
       }
-      if(r == null){
+      if(this.skipListMap != null){
         releaseIndexMemory();
       }
```

Comment:
Yes, On each drop index statement we need to decrease the index overhead from region. I am not sure how IndexManager.removeIndex will help here ? Or maybe I did not understand your question.